### ISSUE TITLE
augmentation: RandomCrop/RandomCrop3D transform_matrix omits the padding offset

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4801]
## Summary

`RandomCrop` and `RandomCrop3D` record a `transform_matrix` that starts from the **padded** canvas and leaves out the pre-crop padding offset. The matrix therefore does not map input pixel coordinates to output pixel coordinates, which is the contract every other 2D geometric augmentation keeps (`CenterCrop`, `RandomResizedCrop`, `PadTo` and `Resize` do map input to output).

`AugmentationSequential` shows the effect. Its keypoint handler adds the padding separately and gets the right answer, but the container's own `.transform_matrix` does not include the padding, so it disagrees with the container's own keypoint output.

## Reproduction (torch 2.14.0, CPU, kornia `main` at 57cd68e6f)

```python
import torch, kornia.augmentation as K

x = torch.zeros(1, 1, 3, 4); x[0, 0, 1, 2] = 1.0          # input pixel (x=2, y=1)
for mode in ("slice", "resample"):
    a = K.RandomCrop((5, 6), padding=1, p=1.0, cropping_mode=mode)
    y = a(x)
    i = int(y[0, 0].argmax())
    print(mode, a.transform_matrix[0].tolist(), "pixel ->", (i % 6, i // 6))
# slice    identity matrix, pixel -> (3, 2)
# resample identity matrix, pixel -> (3, 2)

seq = K.AugmentationSequential(K.RandomCrop((5, 6), padding=1, p=1.0), data_keys=["input", "keypoints"])
y, kp = seq(x, torch.tensor([[[2.0, 1.0]]]))
print(kp.tolist(), (seq.transform_matrix[0] @ torch.tensor([2.0, 1.0, 1.0]))[:2].tolist())
# [[[3.0, 2.0]]]  vs  [2.0, 1.0]

v = torch.zeros(1, 1, 3, 3, 3); v[0, 0, 1, 1, 1] = 1
a = K.RandomCrop3D((5, 5, 5), padding=1, p=1.0); y = a(v)
print(a.transform_matrix[0, :3, 3].tolist(), (y[0, 0] == 1).nonzero().tolist())
# [0.0, 0.0, 0.0]  [[2, 2, 2]]
```

The same happens with `pad_if_needed=True`.

## Expected

The recorded matrix composes the crop with the padding translation `(left, top[, front])`, so `M @ p_input == p_output`. The `RandomCrop3D` docstring currently documents the padded-canvas behaviour as a convention ("Add the left, top, and front padding to an original point before applying this matrix"), and #4791 adds a line to `RandomCrop`. Both become wart lines linking this issue until it is fixed.

Found while reviewing #4791.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).

**Comment by ducha-aiki** [comment]
The same offset appears when the padding comes from `pad_if_needed=True`, with no `padding=` argument, in both cropping modes:

```python
import torch, kornia.augmentation as K
img, pts = torch.zeros(1, 1, 5, 7), torch.tensor([[[1.0, 2.0], [5.0, 4.0]]])
for mode in ("slice", "resample"):
    torch.manual_seed(0)
    seq = K.AugmentationSequential(K.RandomCrop((6, 9), pad_if_needed=True, p=1.0, cropping_mode=mode),
                                   data_keys=["input", "keypoints"])
    _, kp = seq(img, pts)
    via_matrix = (torch.cat([pts[0], torch.ones(2, 1)], 1) @ seq.transform_matrix[0].T)[:, :2]
    print(mode, kp[0].tolist(), via_matrix.tolist())
# slice    [[2.0, 2.0], [6.0, 4.0]] [[0.0, 1.0], [4.0, 3.0]]
# resample [[2.0, 2.0], [6.0, 4.0]] [[0.0, 1.0], [4.0, 3.0]]
```

The keypoint handler adds the `pad_if_needed` padding, `(2, 1)` here, but the matrix leaves it out. A fix should cover both padding sources. #4791 words the wart as "any pre-crop padding" and pins both.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
